Firestore/core/src/firebase/firestore/util/filesystem_posix.cc

```diff
@@ -72,6 +73,15 @@ Status IsDirectory(const Path& path) {
   return Status::OK();
 }
 
+Status Rename(const Path& from_path, const Path& to_path) {
+  if (::rename(from_path.ToUtf8String().c_str(),
+               to_path.ToUtf8String().c_str())) {
+    return Status::FromErrno(errno, from_path.ToUtf8String());
```

Comment:
Out of curiosity, would it make any sense to do `::errno` for consistency with `::rename()` or is that not how this works? :-)

Reply:
Unlike `rename`, which is a function, `errno` is [defined by the standard](https://en.cppreference.com/w/cpp/error/errno) to be a macro. In general C++ scope operators don't apply to macros. The result of the expansion is not defined and it may not be in the global namespace.